Debugging

mmattano opened the pull request:
Removal of white-spaces in calc file
Fix of problem with -i
Addition of requested lines connecting the dots in the growth curves
Temporary addition of a file collecting the specific growth rates and plots visualizing them

The other issues will be fixed ASAP and unit tests will be implemented (working on them)

P.S.: next time I'll create separate pull requests right away, sorry about that

mmattano committed:
fix: Removal of white spaces in the Species column in calc.tsv

White spaces in the "Sample_ID" columns are removed right after import. Also added for df but maybe that could be omitted

mmattano committed:
fix: Remove whitespaces documentation

Added a comment to describe what is happening below

mmattano committed:
fix: Fix command -i, --individual

Label assignment was wrong, changed it for sample name

mmattano committed:
feat: Growth curve plots lines

Added lines connecting the dots

mmattano committed:
feature: temporary collection and visualization of specific growth rates

A new subfolder with a text file containing the specific growth rates for each interval for each sample and plots visualizing them were added

mmattano committed:
tests/fix: possible fix for the CI Linux build/test break issue

According to the reported error message, adding the package "wheel" might fix this issue, will have to be tested.

dmccloskey commented:
@mmattano Is this PR ready for a review?  If so, you need to request me as a reviewer so that I know it is ready.

mmattano committed:
test: Another attempt to pass CI

Following the error message "Failed to build pandas
ERROR: Could not build wheels for pandas which use PEP 517 and cannot be installed directly"
According to what I read on Stackoverflow, downgrading pip might solve that
https://stackoverflow.com/questions/60920252/error-could-not-build-wheels-for-pendulum-which-use-pep-517-and-cannot-be-insta

mmattano commented:
@dmccloskey we can go over the changes but I aim to add tests and corrections for CI today, if it works we could discuss all the things that have been approved by Tadas so far on Monday (or at some other point next week)

mmattano committed:
test: Defining pandas version in setup file

Next attempt to fix the python built

mmattano committed:
fix: improved specific growth rate data export

Changed it from txt to excel file. I also adjusted a tiny return mistake I committed before

mmattano committed:
fix: Volume compensation and background correction

Divided the measurement values by the correlation value instead of multiplying them. Added background correction by keeping the OD600 values for the volume correction samples and subtracting the mean of the background samples for every time step from the samples in that time step

mmattano committed:
test: Attempt to pass circleci by adding xcode

added xcode package as it has been suggested that this might be required to install specifically here in order to make wheel work (which might solve the issue with this specific version of pandas)

mmattano committed:
feat: added plot that plots all species in one with distinct colors

Also adjusted the titles of the existing growth curve plots

mmattano commented:
I still need to find a way to pass circleci and add meaningful unit tests but the rest of the requested changes are ready. There are a few points that I need some clarification on:
-) Volume correction: should the intersect be the intersect or just 1? If it's just 1 then only the effect of volume change will be used for compensation. If it is the actual intersect then all values will be adjusted according to the correlation of the actual first value and the predicted first value. This might be an issue if there are outliers (or just a lot of deviation)
-) Should I add something to prevent messages from popping up during the calculation of the specific growth rates? Right now it prints an error message in the terminal when the concentrations after background correction were negative. This is not an issue for the outcome but I could add something to avoid the message.

mmattano commented:
CI Linux build fixed but unit tests break, will add/adjust them ASAP

mmattano committed:
test: addition of OD600 values to volume correction/background samples

The OD600 values were added to be up to date with the adjusted method

mmattano committed:
test: Circle CI

There was another error, here is another attempt

mmattano committed:
fix: flake8 conform

Changed style

mmattano committed:
build: Circle CI

Next attempt according to https://github.com/pandas-dev/pandas/issues/32045

mmattano committed:
fix: __main__ flake8 conform

Adjusted style

mmattano committed:
build: Circle CI

Moved pandas to the end if other packages are necessary to install it

mmattano committed:
build: Circle CI

Is pandas 1.1.1 really necessary? Let's see what happens with no specification

mmattano committed:
fix: flake8 adjustment

Fixed problem that popped up after initial adjustment according to flake8

dmccloskey commented on tecan_od_analyzer/__main__.py:
```diff
@@ -1,355 +1,507 @@
-from pandas import Series
-from platform import platform
+
+# Additional calls that were listed but not used (remove later)
```
Feel free to remove sooner rather than later.

dmccloskey reviewed (commented):
Do you mind letting me know the specific lines in the different files that you would like a review on?  Due to the large number of changes you introduced for the Flake8 requirements, Git is behaving poorly and making it really hard to see what the "meaningful" changes are.

dmccloskey commented on tecan_od_analyzer/tecan_od_analyzer.py:
```diff
@@ -402,10 +403,28 @@ def compensation_lm(cor_df, df_gr) : #done
+				df_gr_comp["Measurement"].loc[df_gr_comp_index] = df_gr_comp["Measurement"].loc[df_gr_comp_index] - background_mean
+
+		df_gr_comp["Correlation"] = lm_eq[pos][0]*df_gr_comp["time_hours"]+lm_eq[pos][1] # Should this just be 1? Or the normalized initial value
+		#df_gr_comp["Measurement"] = df_gr_comp["Measurement"]*df_gr_comp["Correlation"] # I guess this should be divided rather than multiplied
```
I think the equation needs to be inversed, correct?  So it would be division?

mmattano replied:
I think in that case it needs to be multiplied to get the correlation which is used to divide the values in the next step. Division would also produce an error (division by zero) since we're using the time values and have measurements from time point zero.
If you are referring to the calculation that produces the df_gr_comp["Measurement"] value, I already adjusted everything to the way I describe it in the comments above, guess it just got lost somewhere in the other commits

mmattano committed:
fix: adjusted stepwise growth rate calculation

This was done to avoid the error messages that would usually be displayed if were no positive values available for a well

mmattano committed:
fix: catch warning raised by croissance

Also added a comment to not forget it

mmattano committed:
fix: flake8 adjustment

mmattano committed:
tests: for CircleCI

mmattano committed:
tests: same

mmattano committed:
tests: re-run generate_test_data to account for the new df

mmattano committed:
tests: flake8 and sample outcome adjustment

Also added comments on the bottom for possible extra tests

mmattano committed:
tests: adjusted whitespace test

mmattano committed:
tests: lm_compensations addition of new argument

mmattano committed:
Update test_tecan_od_analyzer.py

mmattano committed:
tests: updated data and calling of dfs

mmattano committed:
Update test_tecan_od_analyzer.py

mmattano committed:
Update test_tecan_od_analyzer.py

mmattano committed:
Update test_tecan_od_analyzer.py

mmattano committed:
fix: added my name to be acountable for issues

mmattano commented:
WIP: can be merged and used. 

Some issues still need to be taken care of:
- other possible unit tests
- error message pops up for the second example data set I received, problem remains to be identified

mmattano committed:
fix: Change png to svg output

Left the original ones in for now as a comment, will take all of that out in the final push

dmccloskey commented on tecan_od_analyzer/tecan_od_analyzer.py:
```diff
@@ -475,7 +478,8 @@ def linear(x, a, b):
 
-    plt.savefig("lm_volume_loss.png", dpi=250)
+    # plt.savefig("lm_volume_loss.png", dpi=250)
+    plt.savefig("lm_volume_loss.svg", dpi=250)
```
`dpi=250` should not be needed with svg output, correct?

dmccloskey commented on tecan_od_analyzer/tecan_od_analyzer.py:
```diff
@@ -475,7 +478,8 @@ def linear(x, a, b):
         df_gr_comp_out = df_gr_comp_out.append(df_gr_comp)
 
-    plt.savefig("lm_volume_loss.png", dpi=250)
+    # plt.savefig("lm_volume_loss.png", dpi=250)
```
I would recommend allowing the user to choose whether they want .png or .svg output by introducing an additional option in the command line arguments.

dmccloskey reviewed (changes requested):
Looking good so far.  My only comments pertain to the figure output.  Once those are done, we should be able to merge.

Also, if you can consolidate the todos into new issues in Jira, that would be great.

mmattano committed:
feat: added svg flag

Added an option to specify if the user wants to export .png or .svg files

mmattano committed:
fix: adjusted flag

mmattano committed:
fix: adjust flag

mmattano committed:
fix: and another attempt

mmattano commented:
Changes implemented, require review, had to try a few times but basically I added a flag and that’s it

dmccloskey reviewed (approved):
This looks good.  Be sure to update the README with the new parameter as this is what users will go to to understand what the different parameters are.  Ping me once that is done and we can merge.

dmccloskey commented:
@mmattano Did you update the README prior to merging?
